Mathematics/Prime.R

```diff
@@ -1,17 +1,21 @@
 # Prime Number Checking in R
-
 isPrime <- function(number) {
-  if (number == 2) {
+  if (number == 2 | number == 3) {
     return(TRUE)
-  }
-  
-  for (x in 2:(number-1)) {
-    if (number %% x == 0) {
-      return(FALSE)
+  } else if (number %% 2 == 0 | number %% 3 == 0){
+    return(FALSE)
+  } else {
+    k <- 1
+    while(6 * k + 1 <= sqrt(number)){
+      if(number %% (6 * k + 1) == 0){
+        return(FALSE)
+      } else if(number %% (6 * k - 1) == 0){
+        return(FALSE)
```

Comment:
What if 6*k-1 == sqrt? Then the check fails and TRUE is returned

Reply:
Yeah you're right- should change that check to be 6k-1 less than or equal to sqrt. Will be a quick fix that I can incorporate tomorrow am and a good catch.